### PR TITLE
chore: re storage update

### PR DESCRIPTION
**Pull request by ganchoradkov**
# Description

- updates `@walletconnect/keyvaluestorage`
- replaces `better-sqlite3` with `lokijs` as dev dep in `sign-client`
- reenables persisted storage tests

Resolves # ([browse](https://github.com/WalletConnect/walletconnect-monorepo/issues) or [create](https://github.com/WalletConnect/walletconnect-monorepo/issues/new/choose))

## How Has This Been Tested?
Tested against prod relay

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

**Review by arein — APPROVED** [review]

**Comment by bkrem** [comment]
@ganchoradkov the dependent PR for this in utils is still open though right? https://github.com/WalletConnect/walletconnect-utils/pull/32

**Comment by ganchoradkov** [comment]
> @ganchoradkov the dependent PR for this in utils is still open though right? [WalletConnect/walletconnect-utils#32](https://github.com/WalletConnect/walletconnect-utils/pull/32)

Yes, doing final tests as some of the issues might appear only in GH actions thus needing to have the PRs open for potential tweaks

**Review by bkrem — APPROVED** [review]

File: packages/core/test/publisher.spec.ts [hunk 1]
```diff
@@ -1,36 +1,33 @@
...
   const logger = pino(getDefaultLoggerOptions({ level: CORE_DEFAULT.logger }));
 
-  let relayer: IRelayer;
+  let core: ICore;
   let publisher: Publisher;
 
   beforeEach(async () => {
-    const core = new Core(TEST_CORE_OPTIONS);
+    if (core) {
+      await disconnectSocket(core);
```
Comment (arein) [inline packages/core/test/publisher.spec.ts:21]: why share the core with each test?

File: packages/sign-client/test/sdk/client.spec.ts [hunk 2]
```diff
@@ -171,20 +174,58 @@ describe("Sign Client Integration", () => {
...
+          await disconnectSocket(beforeClients.A.core);
+          await disconnectSocket(beforeClients.B.core);
+
           deleteClients(beforeClients);
+
+          await new Promise((resolve) => {
+            setTimeout(() => {
+              resolve(true);
+            }, 500);
+          });
```
Comment (arein) [inline packages/sign-client/test/sdk/client.spec.ts]: why?

File: packages/sign-client/test/shared/ws.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { ICore } from "@walletconnect/types";
+import { throttle } from ".";
+
+export async function disconnectSocket(core: ICore) {
+  if (core.relayer.connected) {
+    core.relayer.provider.connect = () => new Promise<void>((resolve) => resolve);
+    await core.relayer.provider.connection.close();
+    throttle(2_000);
```
Comment (arein) [inline packages/sign-client/test/shared/ws.ts]:
```suggestion
    await throttle(2_000);
```

Not clear why this is needed

**Review by arein — APPROVED** [review]

File: packages/core/test/publisher.spec.ts (hunk 1 above)
Comment (ganchoradkov) [reply packages/core/test/publisher.spec.ts:21]: its not shared, its reassigned a few lines later. Just to make sure on the next iteration the sockets are disconnected

**Pull request merged**